tools/webidl_binder.py

```diff
@@ -406,8 +406,19 @@ def render_function(class_name, func_name, sigs, return_type, non_pointer, copy,
   if not js_impl: continue
   implements[name] = [js_impl[0]]
 
+nodeHeight = {}
+for child, parent in implements.iteritems():
```

Comment:
I did leave one line out, but after adding it, I believe this is correct regardless of the order. After one iteration of the loop, all ancestors of child are guaranteed to have a a larger height number than the child, and this is recursively true for each ancestor. If the height of child is later increased, all its ancestors will be readjusted at that time to maintain that invariant. Further, the height of a node never decreases. Therefore, when the loop finishes, all ancestors of a given node should have a larger height number than that node.

The key is in the fact that child will be reprocessed if any of its descendants are discovered later.